Makefile

```diff
@@ -7,15 +8,24 @@ clean-testcache: ## Clean Go test cache
 	go clean -testcache
 
 test:
-	 go test ./...
+	go test ./...
 
 test-cover: clean-testcache ## Run tests with coverage
 	mkdir -p coverage
 	go test -coverprofile=coverage/coverage.out -covermode=atomic .
 	go tool cover -html=coverage/coverage.out -o coverage/coverage.html
-	open coverage/coverage.html
+	if [ -z "$$CI" ]; then open coverage/coverage.html; fi
+
+install-mockgen:
+	@which mockgen || go install go.uber.org/mock/mockgen@v0.5.1
 
 # Need to move mockgen interfaces to their own pkg
-generate: ## Generate mocks
+generate: install-mockgen
 	mkdir -p mocks
 	go generate ./...
+
+install-golangci-lint: 
+	@which golangci-lint || go install github.com/golangci/golangci-lint/cmd/golangci-lint@v2.0.2
```

Comment:
https://golangci-lint.run/welcome/install/ recommend against installing from source, but I think we can either switch to binary installation or use tool pattern, which is easier to manage.